downstairs/src/lib.rs

```diff
@@ -1087,35 +1088,15 @@ async fn proc_task(
              * activated, and then another did (in order to send this thread
              * this signal).
              */
-            new_upstairs_connection = &mut another_upstairs_active_rx => {
-                match new_upstairs_connection {
-                    Err(e) => {
-                        // There shouldn't be a path through the code where we
-                        // close the channel before sending a message through it
-                        // (see [`promote_to_active`]), though [`clear_active`]
-                        // simply drops the active_upstairs tuple - but the only
-                        // place that calls `clear_active` is below when the
-                        // Upstairs disconnects.
-                        //
-                        // We have to bail here though - the Downstairs can't be
-                        // running without the ability for another Upstairs to
-                        // kick out the previous one during activation.
-                        bail!("another_upstairs_active_rx closed during \
-                               proc: {e:?}");
-                    }
-
-                    Ok(new_upstairs_connection) => {
-                        let mut ds = ads.lock().await;
-                        ds.on_new_connection_replacing(
-                            id, new_upstairs_connection);
-                        return Ok(());
-                    }
-                }
+            _ = cancel_io.cancelled() => {
+                info!(log, "proc_loop cancelled; returning now");
+                return Ok(());
```

Comment:
This was returning Error before, but I don't think the difference between error
and Ok resulted in any difference (other than the message).

Reply:
I agree, we basically just want to exit the IO task